cpg_infra/driver.py

```diff
@@ -295,19 +295,42 @@ def resolve_dataset_order(self):
         return graphlib.TopologicalSorter(deps).static_order()
 
     def main(self):
+        # Go through each dataset and instantiate the CPGDatasetInfrastructure class
+        # for that dataset.
         self.setup_datasets()
+
+        # create a bucket and attach accessor members to it. The bucket itself is
+        # created by accessing the property `self.gcp_members_cache_bucket`
+        # This will also have the side effect of creating a cloud resource manager and
+        # identity service
         self.setup_gcp_access_cache_bucket()
+
+        # creates the group metamist-invokers group and gives it invoker permissions
+        # to the cloud run service specified in infrastructure.sample_metadata.gcp.service_name
         self.setup_gcp_metamist_cloudrun_invoker()
+
+        # Create a python registry for storing private python packages
         self.setup_python_registry()
 
+        # Deploy all the assets required for each dataset. Groups, permissions
+        # storage buckets, metamist and hail users etc.
         self.deploy_datasets()
 
-        for plugin in get_plugins().values():
-            plugin(self, self.config).main()
+        for plugin_name, plugin in get_plugins().items():
+            if plugin_name in self.config.plugins_enabled:
+                plugin(self, self.config).main()
```

Comment:
Following on from my comment before, I think loop through the plugins, and if one isn't found raise an Exception.